docs/src/docs/asciidoc/user/resilience.adoc

```diff
@@ -0,0 +1,189 @@
+In this case, after 10 timeouts (configurable), the client will consider the server down to prevent waiting on requests.
+It will perform heartbeats in background to determine when the situation goes back to normal and reconnect if needed.
+Meanwhile, it will rely on the resilience strategy to answer.
+
+*Note:* M&M will trigger an alarm if a server is acting erratically.
+
+It is important to notice that the client won't fallback to the resilience strategy when the caching tier answers.
+It also means that the caching tier might not receive updates from the server and become out of sync.
+
+This is independent of the consistency configured and configurable. You can pick the following strategies:
+
+* Rely on tier forever, so even if the server is officially lost
+* Rely on tier on hiccups. The server will keep the caching tier until it declares the server lost for good
+* Don't rely on tier.
+
+When the server is lost, the client won't try to call it until the heartbeat resumes. A nice feature would be to have a
+ledger pointer in the heartbeat. To tell the client is its out of sync when an answer comes.
+
+=== Active and mirror setup
+
+It works the same as with a single server. Except that during a failover, the client will behaves like if the underlying
+server is having hiccups. Or is down if the failover takes too long.
+
+The new active server will notify the client when ready.
+
+=== Reconnection
+
+When a server goes down and back again. On the same URL. The client will silently reconnect to it. The ledger should allow
+to invalidate entries on the caching tier.
+
+== Loader writer
+
+Cache loader writers have a dedicated resilience strategy. So in case of failure, the resilience strategy will operate and
+the result will be picked up by the Cache resilience strategy.
+
+When using a loader writer, we tend to consider the following:
+
+* We expect the cache never return null. Because the loader will load any missing data. But to enforce that
+in case of server failure, it means the client should wait until the backend is back.
+* We expect that everything cached is saved. So in general an exception should be thrown if we fail to do so.
+
+The default resilience strategy is the following.
+
+* V load(K key): Retry infinitely
+* Map<K, V> loadAll(Iterable<? extends K> keys): Retry infinitely
+* void write(K key, V value): Exception
+* void writeAll(Iterable<? extends Map.Entry<? extends K, ? extends V>> entries): Exception
+* void delete(K key): Exception
+* void deleteAll(Iterable<? extends K> keys): Exception
+
+=== Write behind
+
+The resilience strategy is applied for write behind as well. So make sure it manages errors
+that are asynchronous.
+
+== Interruptions
+
+When waiting on a call to a store, an interruption should allow to get out. It will then probably rely on the resilience
+strategy (I'm not sure about that) or throw an exception right away.
```

Comment:
I don't think it can throw... I think a resilience strategy call might make sense here - but I'd be concerned with how the resultant system might behave in various user applications.  Some people are very ill disciplined about interrupting threads.

Reply:
I'm fine with the resilience strategy. For me, if you interrupt, it should interrupt. We can log when it happens to have a quick diagnosis.